### PR TITLE
Quantizer modulation a la Daala

### PR DESCRIPTION
**Pull request by barrbrain**
Mimic current boosting, at the median value of 15 quantizer steps.
Keep current quantizer values for FRAME_SUBTYPE_P.

AWCY results at [default speed and tune=Psnr](https://beta.arewecompressedyet.com/?job=psnr-4db518c3511d18ab0aca224a0ea295fa18d3f486%402019-02-12T12%3A10%3A45.965Z&job=psnr-a-la-a-daala-rnsp%402019-02-12T14%3A14%3A07.525Z):

|    PSNR | PSNR Cb | PSNR Cr | PSNR HVS |    SSIM | MS SSIM | CIEDE 2000 |
|    ---: |    ---: |    ---: |     ---: |    ---: |    ---: |       ---: |
| -0.8152 | -1.7714 | -1.8240 |  -0.7137 | -1.1781 | -1.1106 |    -1.3330 |

### CODE REVIEW

**Comment by barrbrain** [comment]
AWCY results at [speed 0 and tune=Psnr](https://beta.arewecompressedyet.com/?job=psnr-s0-4db518c3%402019-02-12T14%3A55%3A39.778Z&job=psnr-s0-a-la-daala%402019-02-12T14%3A56%3A19.217Z):

|    PSNR | PSNR Cb | PSNR Cr | PSNR HVS |    SSIM | MS SSIM | CIEDE 2000 |
|    ---: |    ---: |    ---: |     ---: |    ---: |    ---: |       ---: |
| -0.8611 | -1.7823 | -1.7423 |  -0.7831 | -1.1677 | -1.0682 |    -1.6394 |

**Comment by tdaede** [comment]
This is waiting on Tim's version of the same patch.

**Comment by barrbrain** [comment]
I helped rebase Tim's mega-patch atop of recent revisions and reworked my changes to fit.
Tim then rebased and tidied my changes to use as a baseline for splitting the mega-patch.

AWCY results [at default speed and tune=Psychovisual](https://beta.arewecompressedyet.com/?job=master-50269cb681031d6bdd5e7cfa9e55538e288f58ea&job=rate_control7%402019-02-14T01%3A00%3A03.296Z):

|    PSNR | PSNR Cb | PSNR Cr | PSNR HVS |    SSIM | MS SSIM | CIEDE 2000 |
|    ---: |    ---: |    ---: |     ---: |    ---: |    ---: |       ---: |
| -0.8694 | -1.7579 | -1.9049 |  -0.7396 | -1.1682 | -1.0966 |    -1.3843 |

File: src/rate.rs [hunk 1]
```diff
@@ -24,6 +24,31 @@ pub const FRAME_SUBTYPE_B1: usize = 3;
...
+//  coefficients (which depends on the pixel bit depth and the transform
+//  scale), we normalize the quantizer to the equivalent for 8-bit pixels with
+//  orthonormal transforms for the purposes of rate modeling.
+const MQP_Q12: &[i32; FRAME_NSUBTYPES] = &[
+  // TODO: Use a const function once f64 operations in const functions are
+  //  stable.
+  (1.0 * (1 << 12) as f64) as i32,
+  (1.0 * (1 << 12) as f64) as i32,
+  (1.0 * (1 << 12) as f64) as i32,
+  (1.0 * (1 << 12) as f64) as i32
```
Comment (shssoichiro) [inline src/rate.rs]: Are these all intended to be the same?

Comment (barrbrain) [reply src/rate.rs]: Yes, these are intended to intially be the same as the current boosting does not scale with the quantizer. In the near future, we will tune the values in `MQP_Q12` and `DQP_Q57` but for now we just need to capture the current boosting in linear form.

**Pull request merged**